src/controller/scandataexport/execution.go

```diff
@@ -0,0 +1,191 @@
+	"strings"
+	"time"
+)
+
+func init() {
+	task.SetExecutionSweeperCount(job.ScanDataExport, 50)
+}
+
+var Ctl = NewController()
+
+type Controller interface {
+	Start(ctx context.Context, criteria export.Request) (executionId int64, err error)
+	GetExecution(ctx context.Context, executionId int64) (*export.Execution, error)
+	GetExecutions(ctx context.Context, userName string) ([]*export.Execution, error)
+	GetTask(ctx context.Context, executionId int64) (*task.Task, error)
+	DeleteExecution(ctx context.Context, executionId int64) error
+}
+
+func NewController() Controller {
+	return &controller{
+		execMgr:        task.ExecMgr,
+		taskMgr:        task.Mgr,
+		makeCtx:        orm.Context,
+		sysArtifactMgr: systemartifact.Mgr,
+	}
+}
+
+type controller struct {
+	execMgr        task.ExecutionManager
+	taskMgr        task.Manager
+	exportMgr      export.Manager
+	makeCtx        func() context.Context
+	sysArtifactMgr systemartifact.Manager
+}
+
+func (c *controller) GetExecutions(ctx context.Context, userName string) ([]*export.Execution, error) {
+	keywords := make(map[string]interface{})
+	keywords["VendorType"] = job.ScanDataExport
+	keywords[fmt.Sprintf("ExtraAttrs.%s", export.UserNameAttribute)] = userName
+
+	q := new(q2.Query)
+	q.Keywords = keywords
+	execsForUser, err := c.execMgr.List(ctx, q)
+	if err != nil {
+		return nil, err
+	}
+	execs := make([]*export.Execution, 0)
+	for _, execForUser := range execsForUser {
+		execs = append(execs, c.convertToExportExecStatus(ctx, execForUser))
+	}
+	return execs, nil
+}
+
+func (c *controller) GetTask(ctx context.Context, executionId int64) (*task.Task, error) {
+	query := new(q2.Query)
+
+	keywords := make(map[string]interface{})
+	keywords["VendorType"] = job.ScanDataExport
+	keywords["ExecutionID"] = executionId
+	query.Keywords = keywords
+	query.Sorts = append(query.Sorts, &q2.Sort{
+		Key:  "ID",
+		DESC: true,
+	})
+	tasks, err := c.taskMgr.List(ctx, query)
+	if err != nil {
+		return nil, err
+	}
+	if len(tasks) == 0 {
+		return nil, errors.Errorf("No task found for execution Id : %d", executionId)
+	}
+	// for the export JOB there would be a single instance of the task corresponding to the execution
+	// we will hence return the latest instance of the task associated with this execution
+	logger.Infof("Returning task instance with ID : %v", tasks[0].ID)
+	return tasks[0], nil
+}
+
+func (c *controller) GetExecution(ctx context.Context, executionId int64) (*export.Execution, error) {
+	exec, err := c.execMgr.Get(ctx, executionId)
+	if err != nil {
+		logger.Errorf("Error when fetching execution status for ExecutionId: %d error : %v", executionId, err)
+		return nil, err
+	}
+
+	return c.convertToExportExecStatus(ctx, exec), nil
+}
+
+func (c *controller) DeleteExecution(ctx context.Context, executionId int64) error {
+	err := c.execMgr.Delete(ctx, executionId)
+	if err != nil {
+		logger.Errorf("Error when deleting execution  for ExecutionId: %d error : %v", executionId, err)
```

Comment:
"Error when deleting execution for ExecutionId: %d, error: %v"

Reply:
Done